src/components/chat-list/chat-item/chat-item.tsx

```diff
@@ -72,57 +82,57 @@ const ChatItem: React.FC<IChatItemProps> = React.memo(({ chatId }) => {
 
     if (messageToProcess) {
       if (
-        messageToProcess &&
-        (messageToProcess as INormalizedMessage).systemMessageType &&
-        (messageToProcess as INormalizedMessage).systemMessageType !== SystemMessageType.None
+        messageToProcess.systemMessageType &&
+        messageToProcess.systemMessageType !== SystemMessageType.None
       ) {
         return truncate(
           constructSystemMessageText(
             // TODO: replace this logic
-            messageToProcess as INormalizedMessage,
+            messageToProcess,
             t,
-            currentUserId as number,
+            currentUserId,
             lastMessageUserCreator,
           ),
-          {
-            length: 53,
-            omission: '...',
-          },
+          truncateOptions,
         );
       }
 
-      if (chat?.groupChat) {
-        if (isMessageCreatorCurrentUser) {
-          return truncate(`${t('chatFromList.you')}: ${messageToProcess?.text}`, {
-            length: 53,
-            omission: '...',
-          });
+      if (chat.groupChat) {
+        if (isLastMessageCreatorCurrentUser) {
+          return truncate(`${t('chatFromList.you')}: ${messageToProcess.text}`, truncateOptions);
         }
-        return truncate(`${lastMessageUserCreator?.firstName}: ${messageToProcess?.text}`, {
-          length: 53,
-          omission: '...',
-        });
+        return truncate(
+          `${lastMessageUserCreator?.firstName}: ${messageToProcess.text}`,
+          truncateOptions,
+        );
       }
 
-      return truncate(messageToProcess?.text, {
-        length: 53,
-        omission: '...',
-      });
+      return truncate(messageToProcess.text, truncateOptions);
     }
 
     return '';
   }, [
-    chat?.groupChat,
     chatLastMessage,
+    t,
+    chat.groupChat,
     currentUserId,
-    isMessageCreatorCurrentUser,
     lastMessageUserCreator,
-    t,
+    isLastMessageCreatorCurrentUser,
   ]);
 
+  const getMessageDateTime = useMemo(() => {
+    if (!chatLastMessage) return '';
+
+    if (checkIfDatesAreDifferentDate(new Date(chatLastMessage.creationDateTime), new Date())) {
+      return getDayMonthYear(chatLastMessage.creationDateTime);
+    }
+
+    return getShortTimeAmPm(chatLastMessage.creationDateTime).toLowerCase();
+  }, [chatLastMessage]);
```

Comment:
your function depends only on creationDateTime field, why does this method has whole object?

Reply:
Because the React docs state that "every value referenced by a function should also be displayed in an array of dependencies." :)